1998-03-03  Paul Eggert  <eggert@twinsun.com>

	* src/sort.c (xtmpfopen): Open temporary file exclusively, to
	foil a common denial-of-service attack.
	* src/tac.c (save_stdin): Likewise.

--- textutils-1.22/src/sort.c.tmpfile	Sat Jan 18 22:41:55 1997
+++ textutils-1.22/src/sort.c	Fri Mar  6 14:05:30 1998
@@ -321,7 +321,7 @@
   FILE *fp;
   int fd;
 
-  fd = open (file, O_WRONLY | O_CREAT | O_TRUNC, 0600);
+  fd = open (file, O_WRONLY | O_CREAT | O_TRUNC | O_EXCL, 0600);
   if (fd < 0 || (fp = fdopen (fd, "w")) == NULL)
     {
       error (0, errno, "%s", file);
--- textutils-1.22/src/tac.c.tmpfile	Sat Nov  2 18:42:26 1996
+++ textutils-1.22/src/tac.c	Fri Mar  6 14:08:32 1998
@@ -462,7 +462,7 @@
   sprintf (template, "%s/tacXXXXXX", tempdir);
   tempfile = mktemp (template);
 
-  fd = creat (tempfile, 0600);
+  fd = open (tempfile, O_WRONLY | O_CREAT | O_TRUNC | O_EXCL, 0600);
   if (fd == -1)
     {
       error (0, errno, "%s", tempfile);
